pkg/controller/v1alpha1/inferencegraph/raw_ig.go

```diff
@@ -180,6 +211,144 @@ func handleInferenceGraphRawDeployment(cl client.Client, clientset kubernetes.In
+	if apierrors.IsNotFound(err) {
+		clusterRoleAuxiliary := rbacv1.ClusterRole{}
+		rbRoleRef := rbacv1cfg.RoleRef().
+			WithKind("ClusterRole").
+			WithName("system:auth-delegator").
+			WithAPIGroup(clusterRoleAuxiliary.GroupVersionKind().Group)
+		rbSubject := rbacv1cfg.Subject().
+			WithKind("ServiceAccount").
+			WithNamespace(graph.GetNamespace()).
+			WithName(saName)
+		crbApply := rbacv1cfg.ClusterRoleBinding(constants.InferenceGraphAuthCRBName).
+			WithRoleRef(rbRoleRef).
+			WithSubjects(rbSubject)
+
+		_, err = clientset.RbacV1().ClusterRoleBindings().Apply(ctx, crbApply, metav1.ApplyOptions{FieldManager: InferenceGraphControllerName})
+		if err != nil {
+			return errors.Wrapf(err, "fails to apply kserve-inferencegraph-auth-verifiers ClusterRoleBinding for inference graph")
+		}
+	} else {
+		isPresent := false
+		for _, subject := range clusterRoleBinding.Subjects {
+			if subject.Kind == "ServiceAccount" && subject.Name == saName && subject.Namespace == graph.GetNamespace() {
+				isPresent = true
+				break
+			}
+		}
+		if !isPresent {
+			clusterRoleBinding.Subjects = append(clusterRoleBinding.Subjects, rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      saName,
+				Namespace: graph.GetNamespace(),
+			})
+			_, err = clientset.RbacV1().ClusterRoleBindings().Update(ctx, clusterRoleBinding, metav1.UpdateOptions{FieldManager: InferenceGraphControllerName})
+			if err != nil {
+				return errors.Wrapf(err, "fails to bind privileges for auth verification to inference graph")
+			}
+		}
+	}
+
+	return nil
+}
+
+func removeAuthPrivilegesFromGraphServiceAccount(ctx context.Context, clientset kubernetes.Interface, graph *v1alpha1api.InferenceGraph) error {
+	clusterRole, err := clientset.RbacV1().ClusterRoleBindings().Get(ctx, constants.InferenceGraphAuthCRBName, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, "fails to get cluster role binding kserve-inferencegraph-auth-verifiers while deconfiguring inference graph auth")
+	}
+
+	isPresent := false
+	saName := getServiceAccountNameForGraph(graph)
+	for idx, subject := range clusterRole.Subjects {
+		if subject.Kind == "ServiceAccount" && subject.Name == saName && subject.Namespace == graph.GetNamespace() {
+			isPresent = true
+
+			// Remove the no longer needed entry
+			clusterRole.Subjects[idx] = clusterRole.Subjects[len(clusterRole.Subjects)-1]
+			clusterRole.Subjects = clusterRole.Subjects[:len(clusterRole.Subjects)-1]
+			break
+		}
+	}
+
+	if isPresent {
+		_, err = clientset.RbacV1().ClusterRoleBindings().Update(ctx, clusterRole, metav1.UpdateOptions{FieldManager: InferenceGraphControllerName})
+		if err != nil {
+			return errors.Wrapf(err, "fails to remove privileges for auth verification from inference graph")
+		}
+	}
+
+	return nil
+}
+
+func deleteGraphServiceAccount(ctx context.Context, clientset kubernetes.Interface, graph *v1alpha1api.InferenceGraph) error {
```

Comment:
Is the serviceAccount provided by the user? If not, maybe these should be handled by the cleanup function which will delete the owned resources

Reply:
> Is the serviceAccount provided by the user?

Currently, the user cannot specify a Service Account.

> If not, maybe these should be handled by the cleanup function which will delete the owned resources

I'm not sure what you mean here.

Reply:
> I'm not sure what you mean here.

You are deleting it manually, I believe this SA will have owner reference or finalizer that its deletion can be triggered upon the IG deletion.

Reply:
Ah, now I understand.

Yes, the garbage collector can delete it. But this is covering the case when the user would turn off auth. So, the no longer needed SA is deleted despite the IG still exists